presto-main/src/test/java/com/facebook/presto/block/TestDictionaryBlock.java

```diff
@@ -44,6 +48,44 @@ public void testSizeInBytes()
         assertEquals(dictionaryBlock.getSizeInBytes(), dictionaryBlock.getDictionary().getSizeInBytes() + (100 * SIZE_OF_INT));
     }
 
+    @Test
+    public void testNonCachedLogicalBytes()
+    {
+        int numEntries = 10;
+        BlockBuilder blockBuilder = VARCHAR.createBlockBuilder(null, numEntries);
+
+        // Over allocate dictionary indexes but only use the required limit.
+        int[] dictionaryIndexes = new int[numEntries + 10];
+        Arrays.fill(dictionaryIndexes, 1);
+        blockBuilder.appendNull();
+        dictionaryIndexes[0] = 0;
+
+        String string = "";
+        for (int i = 1; i < numEntries; i++) {
+            string += "a";
+            VARCHAR.writeSlice(blockBuilder, utf8Slice(string));
+            dictionaryIndexes[i] = numEntries - i;
+        }
+
+        // A dictionary block of size 10, 1st element -> null, 2nd element size -> 9....9th element size -> 1
+        // Pass different maxChunkSize and different offset and verify if it computes the chunk lengths correctly.
+        Block elementBlock = blockBuilder.build();
+        DictionaryBlock block = new DictionaryBlock(numEntries, elementBlock, dictionaryIndexes);
+        int elementSize = Integer.BYTES + Byte.BYTES;
+
+        long size = block.getRegionLogicalSizeInBytes(0, 1);
+        assertEquals(size, 0 + 1 * elementSize);
+
+        size = block.getRegionLogicalSizeInBytes(0, numEntries);
+        assertEquals(size, 45 + numEntries * elementSize) ;
+
+        size = block.getRegionLogicalSizeInBytes(1, 2);
+        assertEquals(size, 9+8 + 2 * elementSize);
```

Comment:
checkstyle is not happy here

Reply:
I ran mvn clean install, but somehow this got missed.